src/crush/CrushCompiler.cc

```diff
@@ -815,6 +815,42 @@ void CrushCompiler::dump(iter_t const& i, int ind)
     dump(i->children.begin() + j, ind+1); 
 }
 
+int CrushCompiler::adjust_bucket_item_place(iter_t const &i)
+{
+  map<string,set<string> > bucket_items;
+  map<string,iter_t> bucket_itrer;
+  vector<string> buckets;
+  for (iter_t p = i->children.begin(); p != i->children.end(); p++) {
+    if ((int)p->value.id().to_long() == crush_grammar::_bucket) {
+      string name = string_node(p->children[1]);
+      buckets.push_back(name);
+      bucket_itrer[name] = p;
+      for (unsigned q=3; q<p->children.size()-1; q++) {
+        iter_t sub = p->children.begin() + q;
+        string tag = string_node(sub->children[0]);
+        if (tag == "item") {
+          string iname = string_node(sub->children[1]);
+          bucket_items[name].insert(iname);
+        }         
+      }       
+    }     
+  }
+  
+  //adjust the bucket
+  for(unsigned i=0; i<buckets.size(); i++)
+    for(unsigned j=i+1; j<buckets.size(); j++) {
+      if(bucket_items[buckets[i]].count(buckets[j])) {
+        if(bucket_items[buckets[j]].count(buckets[i])) {
+          err << "bucket  '" <<  buckets[i] << "' and bucket '"<< buckets[j] << "' are included each other" << std::endl;
+          return -1; }
+        else {  
+	   std::iter_swap(bucket_itrer[buckets[i]],bucket_itrer[buckets[j]]);
+        } 
```

Comment:
Please clean up upt eh whitespace:

* for ( not for(
* if ( not if(
* } else { not {\nelse {
* avoid long lines > 80 chars
* please include { } on the outer for loop
Thanks!